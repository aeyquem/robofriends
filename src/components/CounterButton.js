import React, { Fragment } from 'react'
import { Component } from 'react';

class CounterButon extends Component {
    constructor(props) {
        super(props);
        this.state = {
            count: 0
        }
    }

    shouldComponentUpdate(nextProps, nextState) {
        if (nextState.count === this.state.count) {
            return false;
        }
        return true;
    }

    updateCount = (event) => {
        this.setState(state => {
            return {
                count: this.state.count + 1
            }
        });
    }

    render() {
        return (
            <div className="tc">
                <button onClick={this.updateCount}>Count: {this.state.count}</button>
            </div>
        )
    }

}

export default CounterButon;