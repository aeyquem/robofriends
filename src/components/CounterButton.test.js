import React from 'react'
import { shallow } from 'enzyme';
import CounterButton from './CounterButton'

it('Displays correctly', () => {
    expect(shallow(<CounterButton />)).toMatchSnapshot();
})

it('Expects state changes', () => {
    const wrapper = shallow(<CounterButton />);
    const instance = wrapper.instance();

    let currentCount = instance.state.count;

    expect(wrapper.state('count')).toEqual(0);
    expect(instance.shouldComponentUpdate({}, { count: currentCount + 1 })).toEqual(true);
    expect(instance.shouldComponentUpdate({}, { count: currentCount })).toEqual(false);
})

it('click updates state', () => {
    const wrapper = shallow(<CounterButton />);
    const instance = wrapper.instance();

    expect(wrapper.state('count')).toEqual(0);
    wrapper.find('button').simulate('click');
    expect(wrapper.state('count')).toEqual(1);
})