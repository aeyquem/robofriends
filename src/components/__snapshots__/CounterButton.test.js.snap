// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays correctly 1`] = `ShallowWrapper {}`;
